modules/apps/foundation/roles/roles-admin-web/src/main/java/com/liferay/roles/admin/web/internal/portlet/RolesAdminPortlet.java

```diff
@@ -282,6 +283,11 @@ public void updateActions(
 
 		long roleId = ParamUtil.getLong(actionRequest, "roleId");
 
+		long groupId = themeDisplay.getScopeGroupId();
+
+		_permissionService.checkPermission(
+			groupId, Role.class.getName(), roleId);
```

Comment:
I went ahead and added the methods to `ResourcePermissionServiceImpl`.  Works great :)